### ISSUE TITLE
Apply permission checks to the list-items endpoint

### ISSUE THREAD

**Issue by sergei-maertens** [GemeenteUtrecht/record-management-app#31]
See [comment](https://github.com/GemeenteUtrecht/record-management-app/pull/27#discussion_r450140932):

Please add an extra permission check to the `rma.destruction.api.FetchListItemsView` endpoint. We want to prevent users logging in with ADFS who are _not assigned to any role_ to be able to view the content of lists, as they expose some details on zaken.

Additionally, I think that depending on the role permission (can_view_case_details), the zaak dict needs to only include the fields that are used in the table, and no extra details. Better to use an allow-list and be explicit about it.

Maybe we can define a mapping from role permission to a list of attributes that are are allowed?

**Comment by annashamray** [comment]
Fixed in https://github.com/GemeenteUtrecht/record-management-app/pull/27/commits/cc9e5355dc02d8a806f1e3c61014c013a60d5def

Actually I don't think this endpoint should show any specific zaak data in any case, because for requesting zaak details we have a separate endpoint. So I limited the response to zaak attributes, which are used in the table, for all roles

Only the author or the assignees of  the destruction list are allowed to make request now.
User without any role will also receive 403 error now.
